crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -1442,6 +1442,9 @@ def non_disruptive_upgrade_compute_nodes(names)
           ::Node.find_node_by_name_or_alias(name)
         end
 
+        # remove nodes which were already upgraded
+        compute_nodes.reject!(&:upgraded?)
```

Comment:
Please also add a `unique` call to remove duplicates from `compute_nodes`. I know that's a different problem, but fix kind of fits in this PR

Reply:
added but I haven't tested this part. please re-review.